sim/common/sod/item_effects/phase_8.go

```diff
@@ -789,13 +788,32 @@ func init() {
 		return character.RegisterAura(core.Aura{
 			Label:    "Ravagane Bladestorm",
 			Duration: time.Second * 9,
+			Icd: &core.Cooldown{
+				Timer:    character.NewTimer(),
+				Duration: time.Second * 8,
+			},
 			OnGain: func(aura *core.Aura, sim *core.Simulation) {
+				if !aura.Icd.IsReady(sim) {
+					return
+				}
+				aura.Icd.Use(sim)
+
+				character.AutoAttacks.CancelAutoSwing(sim)
+				whirlwindSpell.AOEDot().Apply(sim)
+			},
+			OnRefresh: func(aura *core.Aura, sim *core.Simulation) {
+				if !aura.Icd.IsReady(sim) {
+					return
+				}
+				aura.Icd.Use(sim)
+
+				whirlwindSpell.AOEDot().Cancel(sim)
 				character.AutoAttacks.CancelAutoSwing(sim)
-				channelSpell.AOEDot().Apply(sim)
+				whirlwindSpell.AOEDot().Apply(sim)
 			},
 			OnExpire: func(aura *core.Aura, sim *core.Simulation) {
 				character.AutoAttacks.EnableAutoSwing(sim)
-				channelSpell.AOEDot().Cancel(sim)
+				whirlwindSpell.AOEDot().Cancel(sim)
```

Comment:
I know that you mentioned it was sometimes not starting to attack again. Does swapping the order of these fix that?